state.go

```diff
@@ -75,23 +180,31 @@ func (s *State) add(supply *supplyInfo) {
 	s.Lock()
 	defer s.Unlock()
 
-	// Add supply to state
-	s.TotalDelta.Add(s.TotalDelta, supply.Delta)
-	s.TotalReward.Add(s.TotalReward, supply.Reward)
-	s.TotalWithdrawals.Add(s.TotalWithdrawals, supply.Withdrawals)
-	s.TotalBurn.Add(s.TotalBurn, supply.Burn)
+	s.Issuance.GenesisAlloc.Add(s.Issuance.GenesisAlloc, supply.Issuance.GenesisAlloc)
+	s.Issuance.Reward.Add(s.Issuance.Reward, supply.Issuance.Reward)
+	s.Issuance.Withdrawals.Add(s.Issuance.Withdrawals, supply.Issuance.Withdrawals)
+	s.Burn.EIP1559.Add(s.Burn.EIP1559, supply.Burn.EIP1559)
+	s.Burn.Blob.Add(s.Burn.Blob, supply.Burn.Blob)
+	s.Burn.Misc.Add(s.Burn.Misc, supply.Burn.Misc)
+
+	delta := supply.getCalculatedDelta()
+	s.Delta.Add(s.Delta, delta)
```

Comment:
This adds the current block Delta to the one we have in state.

Another idea, is to make Delta dynamic, so as it is being calculated based on the values we have from rest fields in state. (slight pref on this one)